From: Rafael J. Wysocki <rjw@suse.de>
Subject: x86: Fix unknown_nmi_error()
References: bnc#593731
Patch-mainline: no

The behavior of unknown_nmi_error() in SLE11 SP2 is different
from it's behavior in SLE11 SP1 so that it doesn't work as
expected for some customers.  Fix this by making the second half of
unknown_nmi_error() build (and consequently execute) also in the
CONFIG_KDB case. 

Signed-off-by: Rafael J. Wysocki <rjw@suse.de>
---
 arch/x86/kernel/traps.c |    4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

Index: linux-3.0-SLE11-SP2/arch/x86/kernel/traps.c
===================================================================
--- linux-3.0-SLE11-SP2.orig/arch/x86/kernel/traps.c
+++ linux-3.0-SLE11-SP2/arch/x86/kernel/traps.c
@@ -390,8 +390,7 @@ unknown_nmi_error(unsigned char reason,
 		spin_unlock_irqrestore(&kdb_nmi_lock, flags);
 		(void)kdb(LKDB_REASON_ENTER_SLAVE, reason, regs);
 	}
-	return;
-#else
+#endif /* CONFIG_KDB */
 
 	if (notify_die(DIE_NMIUNKNOWN, "nmi", regs, reason, 2, SIGINT) ==
 			NOTIFY_STOP)
@@ -414,7 +413,6 @@ unknown_nmi_error(unsigned char reason,
 		panic("NMI: Not continuing");
 
 	pr_emerg("Dazed and confused, but trying to continue\n");
-#endif /* CONFIG_KDB */
 }
 
 static notrace __kprobes void default_do_nmi(struct pt_regs *regs)
